packages/ember-glimmer/tests/integration/components/closure-components-test.js

```diff
@@ -0,0 +1,671 @@
+import { runDestroy } from 'ember-runtime/tests/utils';
+import { Component } from '../../utils/helpers';
+import { applyMixins, strip } from '../../utils/abstract-test-case';
+import { moduleFor, RenderingTest } from '../../utils/test-case';
+import assign from 'ember-metal/assign';
+import isEmpty from 'ember-metal/is_empty';
+import EventDispatcher from 'ember-views/system/event_dispatcher';
+
+moduleFor('@htmlbars Components test: closure components', class extends RenderingTest {
+  ['@test renders with component helper']() {
+    let expectedText = 'Hodi';
+
+    this.registerComponent('-looked-up', {
+      template: expectedText
+    });
+
+    this.render('{{component (component "-looked-up")}}');
+
+    this.assertText(expectedText);
+
+    this.runTask(() => this.rerender());
+
+    this.assertText(expectedText);
+  }
+
+  ['@test renders with component helper with invocation params, hash']() {
+    this.registerComponent('-looked-up', {
+      ComponentClass: Component.extend().reopenClass({
+        positionalParams: ['name']
+      }),
+      template: '{{greeting}} {{name}}'
+    });
+
+    this.render(strip`
+      {{component (component "-looked-up") "Hodari" greeting="Hodi"}}`
+    );
+
+    this.assertText('Hodi Hodari');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Hodi Hodari');
+  }
+
+  ['@test renders with component helper with curried params, hash']() {
+    this.registerComponent('-looked-up', {
+      ComponentClass: Component.extend().reopenClass({
+        positionalParams: ['name']
+      }),
+      template: '{{greeting}} {{name}}'
+    });
+
+    this.render(strip`
+      {{component (component "-looked-up" "Hodari" greeting="Hodi")
+                  greeting="Hola"}}`
+    );
+
+    this.assertText('Hola Hodari');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Hola Hodari');
+  }
+
+  ['@test updates when component path is bound']() {
+    this.registerComponent('-mandarin', {
+      template: 'ni hao'
+    });
+
+    this.registerComponent('-hindi', {
+      template: 'Namaste'
+    });
+
+    this.render('{{component (component lookupComponent)}}', {
+      lookupComponent: '-mandarin'
+    });
+
+    this.assertText('ni hao');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('ni hao');
+
+    this.runTask(() => this.context.set('lookupComponent', '-hindi'));
+
+    this.assertText('Namaste');
+
+    this.runTask(() => this.context.set('lookupComponent', '-mandarin'));
```

Comment:
This would be become `this.runTask(() => this.context.set('model', { lookupComponent: '-mandarin' }));`